### PR TITLE
Prevent unseteable value for Ref_Payment_ID

### PR DESCRIPTION
**Pull request by yamelsenih**
- Remove null validation for cash
- Prevent unseteable value for Ref_Payment_ID

### CODE REVIEW

File: base/src/org/compiere/model/MPayment.java [hunk 1]
```diff
@@ -2538,16 +2539,12 @@ private void reverseGeneratedPayments() {
...
-						+ "AND EXISTS(SELECT 1 FROM C_Bank b"
-						+ "					INNER JOIN C_BankAccount ba ON(ba.C_Bank_ID = b.C_Bank_ID)"
-						+ "				WHERE ba.C_BankAccount_ID = C_Payment.C_BankAccount_ID"
-						+ "				AND b.BankType = ?)", get_TrxName())
-			.setParameters(getC_Payment_ID(), MPayment.DOCSTATUS_Completed, MBank.BANKTYPE_Bank)
-			.<MPayment>list().stream().forEach(deposit -> {
-				deposit.processIt(MPayment.DOCACTION_Reverse_Correct);
+						"DocStatus = ? "
+						+ "AND IsReceipt = 'Y' "
+						+ "AND EXISTS(SELECT 1 FROM C_Payment p WHERE p.RelatedPayment_ID = C_Payment.C_Payment_ID AND p.Ref_Payment_ID = ?)", get_TrxName())
```
Comment (mckayERP) [inline base/src/org/compiere/model/MPayment.java:2544]: long line

File: base/src/org/spin/process/DepositFromCash.java [hunk 2]
```diff
@@ -77,43 +75,47 @@ protected String doIt() throws Exception {
...
-					//	Count it
-					created.addAndGet(1);
-	  	  		}
-	  	  		//	for Auto Reconcile
-	  	  		if(isAutoReconciled()) {
-	  	  			MBankStatement.addPayment(payment);
-	  	  		}
+  	  		//	Link to Withdrawal
+  	  		Integer referenceId = withdrawalLinkPayments.get(payment.getC_Payment_ID());
+  	  		if(Optional.ofNullable(referenceId).orElse(0).intValue() > 0) {
```
Comment (mckayERP) [inline base/src/org/spin/process/DepositFromCash.java:94]: How about 

            Optional.ofNullable(referenceId).ifPresent(theId -> {
                payment.setRef_Payment_ID(theId);
                payment.saveEx();               
            });

**Review by mckayERP — COMMENTED** [review]
Good change. Added some refactor suggestions.

**Review by mckayERP — APPROVED** [review]

**Comment by yamelsenih** [comment]
Hi @mckayERP,  nice. I will to add it

**Comment by yamelsenih** [comment]
Hi @mckayERP this change was done

**Pull request merged**